modules/seqexec/model/src/main/scala/seqexec/model/ExecutionQueue.scala

```diff
@@ -8,17 +8,19 @@ import cats.implicits._
 import gem.Observation
 import monocle.macros.Lenses
 import seqexec.model.enum.BatchCommandState
+import seqexec.model.enum.BatchExecState
 
 @Lenses
-final case class ExecutionQueue(name:     String,
-                                cmdState: BatchCommandState,
-                                queue:    List[Observation.Id])
+final case class ExecutionQueue(name:      String,
+                                cmdState:  BatchCommandState,
+                                execState: BatchExecState,
```

Comment:
Do you think it is ok to do this? Passing the state here?

Reply:
`execState` is a function of `cmdState` and the execution state of all the sequences identified in `queue`. It is recalculated inside `SeqexecEngine` every time it is needed, so it cannot be out of sync from its dependencies. But if it is set inside `ExecutionQueue`, either it is recalculated at the same time any of its dependencies change, or we agree that its value can only be trusted at the output.
I would feel more comfortable with this alternative: put `ExecutionQueue` back into package `seqexec.server`, without this new `execState`, and create an `ExecutionQueueView` in `seqexec.model` with the extra parameters required by the web server. This View class will have a snapshot of the state of the queue at the output of `SeqexecEngine`.

Reply:
I agree, the way I did it is forcing it. We should do  `ExecutionQueueView` as you suggest